HW02_A_Graph-Fails.Rmd

```diff
@@ -51,36 +68,65 @@ ggplot(data = mpg) +
 ### Graph Fail 3
 Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
 
+**My answer/comment:**
+
+*alpha is for degree of transparency of the points, which ranges from 0(transparent) to 1(opaque). Using "size=2" can make the points bolder.*
+
 Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+
+**My answer/comment**
+
+*The position of the legend can be changed by using theme(legend.position = c()), and I found (0.7.0.9) is a good position for this graph. *
+
+*The legend title can be removed by using theme(legend.title = element_blank())*
+
+*Using them(legend.position= "none") could remove the legend*
+
 ```{r}
 mpg %>% 
 ggplot() + 
-  geom_point(mapping = aes(x = displ, y = hwy, color = class), alpha = 2) + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), size = 2) + 
   theme(legend.direction = "horizontal") + 
-  theme(legend.position = c(5, 40))
+  theme(legend.position = c(0.7, 0.9))+
+  theme(legend.title = element_blank())
+  
 
 ```
 
 ### Graph Fail 4
 I wanted just one smoothing line. Just one line, to show the general relationship here. But that's not happening. Instead I'm getting 3 lines, why and fix it please?
 
+**My answer/comment**
+
+*Wrapping color=drv in ggplot() will apply the color to every layerl, which is why there were three seperate lines. To have one line, I moved the color=drv to the geom_point. This way it's only applied to the points.*
+
+*I also noticed an interesting fact that the block of codes below would fail with an error message saying "%>% cannot be found" and wont if I didnt run the library("magrittr") in the very first r code block in this file.* 

```

Comment:
This is an interesting finding! Did you also have the same issue when using pipe for other graphs?

Reply:
Yep, not just for pipe, but in same rmd file, sometimes it says ggplot cannot be found, without me re-running the block where specify library("ggplot2"). The solution might be on the top Run button to run all chunks at once?